pkg/cache/sql/db/client.go

```diff
@@ -274,6 +276,22 @@ func (c *Client) Upsert(tx TXClient, stmt *sql.Stmt, key string, obj any, should
 	return tx.StmtExec(tx.Stmt(stmt), key, objBytes, dataNonce, kid)
 }
 
+func (c *Client) UpsertLabels(tx TXClient, stmt *sql.Stmt, key string, obj any, shouldEncrypt bool) error {
+	k8sObj, ok := obj.(*unstructured.Unstructured)
+	if !ok {
+		logrus.Debugf("UpsertLabels: Error?: Can't convert obj into an unstructured thing.")
+		return nil
+	}
+	incomingLabels := k8sObj.GetLabels()
+	for k, v := range incomingLabels {
+		err := tx.StmtExec(tx.Stmt(stmt), key, k, v)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
```

Comment:
db/client was designed to be a wrapper around relatively low-level database operations, and should in my eyes be kept independent from logic which depends on the exact table structure ("application concerns").

I think this code belongs to the same place where the bulk of the logic for labels goes, see the discussion about store below.